### DISCUSSION TITLE
BirdNET-stream: BirdNET-Analyzer based, BirdNET-Pi inspired application for any Linux computer

### DISCUSSION THREAD

**Discussion by samuelortion** [mcguirepr89/BirdNET-Pi#583] [General]
Hello,
As I had no mic for my Raspberry Pi, but a homemade jack microphone and parabola working with an old laptop, I tried to use BirdNET-Pi on a Debian Bullseye x86_64 machine, with some BirdNET-Pi modification. 
Unfortunately, the webpage worked but no sound were recorded.

In order to make this system work, I rewrite the application, with great inspiration from BirdNET-Pi. An now this is working on my system.

Maybe I could help for BirdNET-Pi to be working on any x86_64 computer. I think there is not that much to do to adapt the code.

Bitwise, here is the repository of [BirdNET-stream](https://github.com/UncleSamulus/BirdNET-stream).

BirdNET-stream has a lot less features than BirdNET-Pi yet, but I hope it could help those who don't own a raspberry with microphone, to run BirdNET continuously as well.

It is not mature enough right now, but I am working on It.

**Comment by mcguirepr89** [comment]
## Super cool!!

**Comment by matthew73210** [comment]
#211 
I've just referenced this other issue as the author of this issue is currently doing work on VM / docker support

**Comment by pr0j3ctx** [comment]
This would be great. Today I tried to install Birdnet-Pi into LXC Container with Ubuntu 22.04.
It's as I expected. It doens't work.  
I want to use the audio stream from my Unifi Camera, so running this virtually in the LXC would be great.

I have tried the steps from https://github.com/mcguirepr89/BirdNET-Pi/issues/204#issuecomment-1119949484 but this doesn't work for me. If I put tensorflow and tflite to the requirements I have issues with python imports.

```
Apr 20 14:56:13 birdnet python3[1106208]: Traceback (most recent call last):
Apr 20 14:56:13 birdnet python3[1106208]:   File "/usr/local/bin/server.py", line 25, in <module>
Apr 20 14:56:13 birdnet python3[1106208]:     import tflite.interpreter as tflite
Apr 20 14:56:13 birdnet python3[1106208]: ModuleNotFoundError: No module named 'tflite.interpreter'
Apr 20 14:56:13 birdnet python3[1106208]: During handling of the above exception, another exception occurred:
Apr 20 14:56:13 birdnet python3[1106208]: Traceback (most recent call last):
Apr 20 14:56:13 birdnet python3[1106208]:   File "/usr/local/bin/server.py", line 27, in <module>
Apr 20 14:56:13 birdnet python3[1106208]:     from tensorflow import lite as tflite
Apr 20 14:56:13 birdnet python3[1106208]: ModuleNotFoundError: No module named 'tensorflow'
```

If I tried this wheel file https://github.com/maxogden/BirdNET-Pi/blob/main/tflite_runtime-2.6.0-cp39-cp39-linux_x86_64.whl instead. It tells me that's not supported for my platform.

```
ERROR: tflite_runtime-2.6.0-cp39-cp39-linux_x86_64.whl is not a supported wheel on this platform.
user@birdnet:~/BirdNET-Pi/scripts$ uname -m
x86_64
```

Any hints?
